inst/private/doctest_collect.m

```diff
@@ -26,7 +26,7 @@
     type = 'texinfo';
   elseif (exist (what) == 3)  % .oct/.mex
     [~, what, ~] = fileparts (what);  % strip extension if present
-    type = 'function';                % then access like any function
+    type = 'octfile';                 % then access like any function
```

Comment:
in my patch, I made a distinction b/w `doctest foo` and `doctest foo.oct`: whereby the former tests only the function called foo and the latter tests all functions in `foo.oct` (including `foo` itself).  Thoughts?

Reply:
Yes, I like your approach better.